### ISSUE TITLE
plot_date() does not work with x values of type pandas.Timestamp (pandas version 0.15.0)?

### ISSUE THREAD

**Issue by pybokeh** [matplotlib/matplotlib#3727]
Please see my ipython notebook gist:
http://nbviewer.ipython.org/gist/pybokeh/f2aab3aafdc287d02bb8

Basically, if I try to plot_date() using x values of type datetime, I get weird errors.  But if I convert the x values to type datetime.date, then it plots ok.

EDIT: the only difference between In[3] and In[4] is I used list comprehension in the plot_date() method in In[4] to convert the x values to type datetime.date.

**Comment by petehuang** [comment]
All gist links are broken. Is there another example that we can use to confirm the fix provided?

**Comment by jorisvandenbossche** [comment]
This issue can actually just be closed I think. The actual issue reported here is closed by a PR in pandas in version 0.15.1(so only pandas 0.15.0 has this problem), and the larger issue (better datetime support in matplotlib) is covered by #1097

**Comment by pybokeh** [comment]
I re-ran my old examples and confirmed that with latest version of mpl and pandas, all works ok.  Closing this issue.

**Comment by liubenyuan** [comment]
It seems like this problem happens in pandas 0.21.0.

code to reproduce this error
```python
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from matplotlib import dates

t = np.arange(10)
ts = pd.datetime(2017, 11, 16) + pd.to_timedelta(t, 's')
df = pd.DataFrame(t, index=ts)

fig, ax = plt.subplots()
ax.plot(df)

# format time axis
hfmt = dates.DateFormatter('%m/%d %H:%M')
ax.xaxis.set_major_formatter(hfmt)
fig.autofmt_xdate()
plt.show()

```

The trace
```python
Traceback (most recent call last):

  File "D:\Anaconda3\lib\site-packages\IPython\core\formatters.py", line 332, in __call__
    return printer(obj)

  File "D:\Anaconda3\lib\site-packages\IPython\core\pylabtools.py", line 237, in <lambda>
    png_formatter.for_type(Figure, lambda fig: print_figure(fig, 'png', **kwargs))

  File "D:\Anaconda3\lib\site-packages\IPython\core\pylabtools.py", line 121, in print_figure
    fig.canvas.print_figure(bytes_io, **kw)

  File "D:\Anaconda3\lib\site-packages\matplotlib\backend_bases.py", line 2208, in print_figure
    **kwargs)

  File "D:\Anaconda3\lib\site-packages\matplotlib\backends\backend_agg.py", line 507, in print_png
    FigureCanvasAgg.draw(self)

  File "D:\Anaconda3\lib\site-packages\matplotlib\backends\backend_agg.py", line 430, in draw
    self.figure.draw(self.renderer)

  File "D:\Anaconda3\lib\site-packages\matplotlib\artist.py", line 55, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)

  File "D:\Anaconda3\lib\site-packages\matplotlib\figure.py", line 1295, in draw
    renderer, self, artists, self.suppressComposite)

  File "D:\Anaconda3\lib\site-packages\matplotlib\image.py", line 138, in _draw_list_compositing_images
    a.draw(renderer)

  File "D:\Anaconda3\lib\site-packages\matplotlib\artist.py", line 55, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)

  File "D:\Anaconda3\lib\site-packages\matplotlib\axes\_base.py", line 2399, in draw
    mimage._draw_list_compositing_images(renderer, self, artists)

  File "D:\Anaconda3\lib\site-packages\matplotlib\image.py", line 138, in _draw_list_compositing_images
    a.draw(renderer)

  File "D:\Anaconda3\lib\site-packages\matplotlib\artist.py", line 55, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)

  File "D:\Anaconda3\lib\site-packages\matplotlib\axis.py", line 1133, in draw
    ticks_to_draw = self._update_ticks(renderer)

  File "D:\Anaconda3\lib\site-packages\matplotlib\axis.py", line 974, in _update_ticks
    tick_tups = list(self.iter_ticks())

  File "D:\Anaconda3\lib\site-packages\matplotlib\axis.py", line 921, in iter_ticks
    for i, val in enumerate(majorLocs)]

  File "D:\Anaconda3\lib\site-packages\matplotlib\axis.py", line 921, in <listcomp>
    for i, val in enumerate(majorLocs)]

  File "D:\Anaconda3\lib\site-packages\matplotlib\dates.py", line 538, in __call__
    dt = num2date(x, self.tz)

  File "D:\Anaconda3\lib\site-packages\matplotlib\dates.py", line 441, in num2date
    return _from_ordinalf(x, tz)

  File "D:\Anaconda3\lib\site-packages\matplotlib\dates.py", line 256, in _from_ordinalf
    dt = datetime.datetime.fromordinal(ix).replace(tzinfo=UTC)

OverflowError: Python int too large to convert to C long

<matplotlib.figure.Figure at 0xebecc88>
```

**Comment by jorisvandenbossche** [comment]
See https://github.com/matplotlib/matplotlib/issues/9610 (short story is that pandas no longer automatically registers its unit converters starting from 0.21.0), but we will fix this in 0.21.1 (https://github.com/pandas-dev/pandas/issues/18301)
